apm-agent-core/src/main/java/co/elastic/apm/agent/bci/OsgiBootDelegationEnabler.java

```diff
@@ -37,18 +34,25 @@
  * </p>
  */
 public class OsgiBootDelegationEnabler implements LifecycleListener {
-    private static final List<String> bootdelegationNames = Arrays.asList("org.osgi.framework.bootdelegation", "atlassian.org.osgi.framework.bootdelegation");
     private static final String APM_BASE_PACKAGE = "co.elastic.apm.agent.*";
+    // see https://confluence.atlassian.com/jirakb/using-javaagent-with-jira-790793295.html#UsingjavaagentwithJIRA-Resolution
+    private static final String ATLASSIAN_BOOTDELEGATION = "META-INF.services,com.yourkit,com.singularity.*,com.jprofiler," +
+        "com.jprofiler.*,org.apache.xerces,org.apache.xerces.*,org.apache.xalan,org.apache.xalan.*,sun.*,com.sun.jndi.*,com.icl.saxon," +
+        "com.icl.saxon.*,javax.servlet,javax.servlet.*,com.sun.xml.bind.*,co.elastic.apm.agent.*";
 
     @Override
     public void start(ElasticApmTracer tracer) {
-        for (String bootdelegationName : bootdelegationNames) {
-            final String systemPackages = System.getProperty(bootdelegationName);
-            if (systemPackages != null) {
-                System.setProperty(bootdelegationName, systemPackages + "," + APM_BASE_PACKAGE);
-            } else {
-                System.setProperty(bootdelegationName, APM_BASE_PACKAGE);
-            }
+        // may be problematic as it could override the defaults in a properties file
+        appendToSystemProperty("org.osgi.framework.bootdelegation", APM_BASE_PACKAGE);
+        appendToSystemProperty("atlassian.org.osgi.framework.bootdelegation", ATLASSIAN_BOOTDELEGATION);
```

Comment:
Maybe the correct way to go with this one is- if already exists- add only our package, else add the entire thing